iib/workers/dogpile_cache.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import threading
+
+from celery import current_task
+from dogpile.cache import make_region
+
+from iib.workers.config import get_worker_config
+
+
+def generate_cache_key(fname, *args, **kwargs):
+    """Generate key that is used in dogpile cache."""
+    namespace = dogpile_namespace.get_dogpile_namespace()
+    arguments = "_".join(
+        [str(arg) for arg in args] + [f"{kwarg}={kwargs[kwarg]}" for kwarg in kwargs]
+    )
+    return f"{namespace}_{fname}_{arguments}"
+
+
+def create_dogpile_region():
+    """Load setting from config and creates dogpile region."""
+    conf = get_worker_config()
+
+    return make_region().configure(
+        conf.iib_dogpile_backend,
+        expiration_time=conf.iib_dogpile_expiration_time,
+        arguments=conf.iib_dogpile_arguments,
+    )
+
+
+class DogpileNamespace:
```

Comment:
Could you please explain why a namespace is required? Based on my understanding, if we use `cache_on_arguments` as I describe in another comment, it will use [get_or_create](https://dogpilecache.sqlalchemy.org/en/latest/api.html#dogpile.cache.region.CacheRegion.get_or_create). The documentation for `get_or_create` says the following:
> Whether or not the function is used depends on if the dogpile lock can be acquired or not. If it can’t, it means a different thread or process is already running a creation function for this key against the cache. When the dogpile lock cannot be acquired, the method will block if no previous value is available, until the lock is released and a new value available. If a previous value is available, that value is returned immediately without blocking.

Since locking is in place automatically and we only want to cache skopeo inspect calls that use a digest for now, it should be fine to use the default behavior.

Reply:
To further clarify, I think sharing the same cache for all requests is fine since we are only caching something that is sure to be a stable value for the duration of `conf.iib_dogpile_expiration_time`.